### ISSUE TITLE
Assets not listing when using https

### ISSUE THREAD

**Issue by licks0re** [grokability/snipe-it#12282]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Hello people,

I have an issue I can't figure out.
I'm running a self hosted version of snipeit.

-Clean ubuntu install
-Latest version from git

-Cloudflare in front
*no https forcing

-2 vhosts in apache, http and https. 

-.env states `http://domain.ltd`  as  APP_URL

As soon as I switch on the https version, I cannot see any item listing. Assets, accessories, etc, all will display the message : 
`No matching records found`

I've been testing all I can, but I'm reaching the end of what I can do as I'm writing, I really don't see what could cause that.
Debug mode won't show any error.

Any ideas, thoughts which could help is welome :)

Thanks!
Guillaume

### Reproduction steps

1. git clone
2. run install script
3. log in, configure preflight and add user
4. add an asset or acessory
5. see your newly created object in the list.
6. switch to https
7. No listing, you can see the number of accessories on the big icon, but nothing is listing in the list.
8. When doing a search, item will display *but* it's back to http.
...

### Expected behavior

https list items like in the http version

### Screenshots

https:
![image](https://user-images.githubusercontent.com/9083552/209696849-1759010f-5224-46db-be5e-00e048b53178.png)

http
![image](https://user-images.githubusercontent.com/9083552/209696542-7fc7123d-ae92-45f0-b275-0608c49cbbee.png)

### Snipe-IT Version

6.0.14

### Operating System

ubuntu

### Web Server

apache

### PHP Version

8.1.2

### Operating System

win 10

### Browser

chrome

**Comment by iangozer** [comment]
and when you change .env to https://domain.ltd does it work?

**Comment by licks0re** [comment]
> and when you change .env to https://domain.ltd does it work?

No, it does the same.

**Comment by iangozer** [comment]
Try changing the .env file to the IP or change your hosts file top point to the IP. This is by-passing Cloudflare and should prove if it is an issue directly with snipe-it or a configuration with Cloudflare.

**Comment by licks0re** [comment]
Hello iangozer,

Thanks for your suggestions. I did try that but no luck. I ended up starting from scratch, without cloudflare in front.

I set up http : no prob
I set up https : no prob
I forced the https through the .htaccess file = worked

I re-enabled cloudflare and I had a "too many redirections" error, which I fixed with setting the https mode as "Strict" and not "flexible" (in cloudflare config).

Now, everything is working, so I'll close the issue, thanks again for your help here.

But before closing, I saw in my logs that the server was presenting some mix content. At the end, I think it was not a snipeit problem but some residual bad configuration in my apache vhosts configuration. Unfortunately, as I restarted from scratch with new vm, I can't drill more into this.
